src/main/java/org/opencadc/tap/impl/AuthenticatorImpl.java

```diff
@@ -57,43 +72,27 @@ public Subject augment(Subject subject)
         for (Principal principal : subject.getPrincipals()) {
             if (principal instanceof BearerTokenPrincipal) {
                 BearerTokenPrincipal tp = (BearerTokenPrincipal) principal;
-
-                try {
-                    JsonObject authData = null;
-                    boolean success = false;
-                    for (int i = 1; i < 5; i++) {
-                        try {
-                            authData = getTokenInfo(tp.getName());
-                            success = true;
-                        } catch (IOException e) {
-                            log.warn(e);
-                            log.warn("IOException while getting info Gafaelfawr, retrying");
-                        }
+                TokenInfo tokenInfo = null;
+
+                for (int i = 1; i < 5 && tokenInfo == null; i++) {
+                    try {
+                        tokenInfo = getTokenInfo(tp.getName());
+                    } catch (IOException|InterruptedException e) {
+                        log.warn("Exception thrown while getting info from Gafaelfawr");
+                        log.warn(e);
                     }
+                }
 
-                    if (!success) {
-                        // Try one more time to throw an accurate exception.
-                        authData = getTokenInfo(tp.getName());
-                    }
-
-                    String username = authData.getAsJsonPrimitive("username").getAsString();
-                    int uid = authData.getAsJsonPrimitive("uid").getAsInt();
-
-                    X500Principal xp = new X500Principal("CN=" + username);
+                if (tokenInfo != null) {
```

Comment:
I think it would be better to have an else for this if so that we explicitly log some sort of error when we've exhausted the retries and given up.